app/controllers/ileQuery/IleQueryController.scala

```diff
@@ -115,7 +115,11 @@ class IleQueryController @Inject()(
 
         ducrResult.orElse(mucrResult).getOrElse(loadingPageResult)
 
-      case _: UcrNotFoundResponseExchangeData => Ok("UCR not found")
+      case response: UcrNotFoundResponseExchangeData =>
+        response.ucrBlock match {
+          case Some(UcrBlock(ucr, _)) => Redirect(controllers.ileQuery.routes.ConsignmentNotFoundController.displayPage(ucr))
```

Comment:
Did we need a whole new end-point and controller here?  Could we not just `Ok(consignmentNotFound(ucr))` (assuming the new not-found page was imported into this controller).

Reply:
IMHO, this improves readability. The `IleQueryController` has already four other pages, namely  ile_query, loading_screen, ile_query_ducr_response, and ile_query_mucr_response.

Reply:
IMO it makes the whole process even more complicated.  It's already bad enough with the redirects on checking the progress of the query.  The pages could be quite straight forward - 
1. start page (form)
2. in-progress page
3. two versions of "success"
4. failure (not found) page

Reply:
I will remove the additional controller.